### PR TITLE
C963 apply

### PR DESCRIPTION
**Pull request by BSick7**
This makes changes necessary to support a CI workflow for Terraform.
- Added `plan` command.
- Added `apply` command.
- Replaced `--version=auto` for `--version=next-patch` in `modules publish`.
- Added `--version=next-build` in `modules publish` to use `next-patch` and `+build` suffix using the short commit sha.

The `up` command was refactored to make the code for running Terraform reusable across `plan`, `apply`, and `up`.

### TODO
- [x] Merge https://github.com/nullstone-io/go-api-client/pull/31
- [x] Update go-api-client dependency + go mod tidy

### CODE REVIEW

File: cmd/apply.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	"gopkg.in/nullstone-io/nullstone.v0/runs"
+	"os"
+	"strings"
+)
+
+var Apply = func() *cli.Command {
+	return &cli.Command{
+		Name:      "Apply",
+		Usage:     "Runs an apply with optional auto-approval",
+		UsageText: "nullstone plan [--stack=<stack-name>] --block=<block-name> --env=<env-name> [options]",
```
Comment (ssickles) [inline cmd/apply.go:18]: Should this usage text say `nullstone plan`? Shouldn't it be `nullstone apply`?

File: cmd/apply.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		Flags: []cli.Flag{
+			StackFlag,
+			BlockFlag,
+			EnvFlag,
+			&cli.BoolFlag{
+				Name:    "wait",
+				Aliases: []string{"w"},
+				Usage:   "Stream the Terraform logs while waiting for Nullstone to run the plan.",
```
Comment (ssickles) [inline cmd/apply.go:26]: Swap the word `plan` for `apply`.

File: cmd/apply.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			},
+			&cli.BoolFlag{
+				Name:  "auto-approve",
+				Usage: "Auto-approve any changes made in Terraform",
+			},
+			&cli.StringSliceFlag{
+				Name:  "var",
+				Usage: "Set variable values when issuing `plan`",
```
Comment (ssickles) [inline cmd/apply.go:34]: Swap the word `plan` for `apply`.

File: cmd/apply.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+			},
+			&cli.StringFlag{
+				Name:  "module-version",
+				Usage: "Use a specific module version to run the plan.",
```
Comment (ssickles) [inline cmd/apply.go:38]: Swap the word `plan` for `apply`.

File: cmd/apply.go (hunk 1 above)
Comment (ssickles) [inline cmd/apply.go:18]: Shouldn't this say `nullstone apply` instead of `nullstone plan`?

File: runs/stream_logs.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	innerCtx, cancelFn := context.WithCancel(ctx)
+	defer cancelFn()
+
+	fmt.Fprintln(os.Stdout, "Waiting for logs...")
+	client := api.Client{Config: cfg}
+	msgs, err := client.LiveLogs().Watch(innerCtx, workspace.StackId, newRun.Uid)
+	if err != nil {
+		return err
+	}
+	runCh := pollRun(innerCtx, cfg, workspace.StackId, newRun.Uid, time.Second)
```
Comment (ssickles) [inline runs/stream_logs.go:27]: I'm not understanding the purpose of this. If we are receiving the logs via websockets, why do we need to poll to retrieve runs?

**Review by ssickles — CHANGES_REQUESTED** [review]
My last comment in StreamLogs is the only reason for not approving this. I'm sure I just don't fully understand. Let's huddle on this.

File: cmd/apply.go (hunk 1 above)
Comment (BSick7) [reply cmd/apply.go:18]: Done.

File: cmd/apply.go (hunk 2 above)
Comment (BSick7) [reply cmd/apply.go:26]: Done.

File: cmd/apply.go (hunk 3 above)
Comment (BSick7) [reply cmd/apply.go:34]: Done.

File: cmd/apply.go (hunk 4 above)
Comment (BSick7) [reply cmd/apply.go:38]: Done.

File: cmd/apply.go (hunk 1 above)
Comment (BSick7) [reply cmd/apply.go:18]: Done.

File: runs/stream_logs.go (hunk 5 above)
Comment (BSick7) [reply runs/stream_logs.go:29]: I added a TODO to replace this with an EOF message pumped through the live logs.

**Pull request merged**